File: src/com/lucy/ocserver/httpHandlers/PSSstatusHandler.java
package com.lucy.ocserver.httpHandlers;

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;

import java.io.IOException;
import java.util.Map;

public class PSSstatusHandler implements HttpHandler {
    private long capacity = 1;
    private long stored = 0;
    private boolean reqMaintenance = false;


    @Override
    public void handle(HttpExchange exchange) throws IOException {
        Map<String, String> parameters = RootHandler.queryToMap(exchange.getRequestURI().getQuery());

        // update values
        if (parameters.containsKey("capacity")) {
            capacity = Long.valueOf(parameters.get("capacity"));
            System.out.println("Set capacity to " + capacity);
        }
        if (parameters.containsKey("stored")) {
            stored = Long.valueOf(parameters.get("stored"));
            System.out.println("Set stored EU to " + stored);
        }
        if (parameters.containsKey("maintenance")) {
            reqMaintenance = parameters.get("maintenance").equals("true");
            System.out.println("Set maintenance to " + reqMaintenance);
        }

        String pssStatus = stored + "/" + capacity + "/" + reqMaintenance;
        RootHandler.sendResponse(exchange, pssStatus);
    }
}

File: src/com/lucy/ocserver/httpHandlers/PssStatusHandler.kt
package com.lucy.ocserver.httpHandlers

import com.google.gson.Gson
import com.lucy.ocserver.EU
import com.lucy.ocserver.objects.PowerPack
import com.lucy.ocserver.objects.PowerPackJson
import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import java.io.IOException

class PssStatusHandler : HttpHandler {
    private val powerPacks = mutableMapOf(PowerPackLocation.OldBase to PowerPack(1, 0, false))

    @Throws(IOException::class)
    override fun handle(exchange: HttpExchange) {
        val parameters = RootHandler.queryToMap(exchange.requestURI.query)
        handleUpdate(parameters)

        val powerPacksJson: MutableMap<PowerPackLocation, PowerPackJson> = mutableMapOf()
        for ((location, pack) in powerPacks) {
            powerPacksJson[location] = pack.getJsonObject()
        }
        val json = Gson().toJson(powerPacksJson)
        RootHandler.sendResponse(exchange, json)
    }

    private fun handleUpdate(parameters: Map<String, String>) {
        val name = parameters["name"] ?: return
        val capacity: EU = parameters["capacity"]?.toLong() ?: return
        val stored: EU = parameters["stored"]?.toLong() ?: return
        val maintenance = parameters["maintenance"] == "true"
        val location = PowerPackLocation.valueOf(name)

        var powerPack = powerPacks[location]
        if (powerPack != null) {
            powerPack.capacity = capacity
            powerPack.set(stored)
            powerPack.maintenance = maintenance
        } else {
            powerPack = PowerPack(capacity, stored, maintenance)
        }

        powerPacks[location] = powerPack
    }
}

enum class PowerPackLocation {
    OldBase, NewBase, Chem
}